### ISSUE TITLE
chalice support python 3.9 as host python

### ISSUE THREAD

**Issue by kapilt** [aws/chalice#1707]
issue and solution documented in https://github.com/aws/chalice/issues/1696
specifically re issue https://github.com/aws/chalice/issues/1696#issuecomment-817369468
specifically re solution, pass --python-version 38 to pip (or alternative lambda runtime version).

tldr, chalice doesn't properly package cross python version dependencies.

**Comment by jamesls** [comment]
Yeah in general we've always recommended that your local version of python matches the version of python you want to run on lambda.  It's easy to accidentally use something in 3.9 that either doesn't exist of works differently in 3.8 or whatever Lambda version you're running.  I think if we were going to support this, I'd like to have this be an explicit config option you have to set in your config file (`lambda_python_version`).  Right now we try to make a best effort attempt and warn you if you have a version mismatch.

**Comment by manmartgarc** [comment]
Is it possible to set the lambda version via the chalice config file?
